### ISSUE TITLE
iOS: per-subscription mute

### ISSUE THREAD

**Issue by binwiederhier** [binwiederhier/ntfy#278]

**Comment by binwiederhier** [comment]
https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_usernotifications_filtering

**Comment by binwiederhier** [comment]
"Dear Developer,

Thank you for your interest in the Notification Service Extension filtering entitlement, and thank you for the detailed explanation of your use case. 



We are unable to approve your request at this time. On iOS, we generally expect apps to make the determination of whether they need to alert the user on the server, not the client, in the interest of battery life. 



See [Choosing Background Strategies for Your App](https://developer.apple.com/documentation/backgroundtasks/choosing_background_strategies_for_your_app?language=objc) for alternative APIs that could help with your usecase, such as Background Push. 


You may still use the Notification Service Extension without the filtering entitlement. As long as your Notification Service Extension runtime yields a visible notification at the end, it can [remove previously posted notifications](https://developer.apple.com/documentation/usernotifications/unusernotificationcenter/1649500-removedeliverednotificationswith?language=objc). 



You can also consider using a lower [interruptionLevel](https://developer.apple.com/documentation/usernotifications/unnotificationinterruptionlevel?changes=la&language=objc) on iOS 15 and later.  



Best Regards,

Andy (iOS Battery Life)"

**Comment by binwiederhier** [comment]
Based on this response from Apple, i don't believe there is a way to implement the "mute topic" feature or the "min priority" feature without tracking the user server side, which I don't want to do. 

I'll le

**Comment by binwiederhier** [comment]
"
Good morning Andy,

Thank you for the swift reply.

> On iOS, we generally expect apps to make the determination of whether they need to alert the user on the server, not the client, in the interest of battery life.

That makes a lot of sense for traditional messaging apps. However, ntfy is a pubsub system that solely operates on topics (Firebase topics, and others). Server side, I do not track the Firebase token or APNs token at all, meaning I cannot distinguish users server side. A topic may have many subscribers (users), and sending exclusively to topics with multiple users means I cannot mute based on user, because different users may have different mute-preferences.

TLDR: ntfy uses topics with multiple subscribers (users), so communication is 1:n. Muting server side is hence not possible.

I should have included this in the original email. I ask you kindly and with the utmost respect to reconsider your evaluation.

Best regards,"

**Comment by MrOwen** [comment]
Did Apple ever offer a response after your reply?

**Comment by binwiederhier** [comment]
No answer, sadly.

**Comment by ksurl** [comment]
This is unfortunate. Mute is pretty nice on discord.

**Comment by RezzZ** [comment]
> This is unfortunate. Mute is pretty nice on discord.

not only discord, also great feature on apps like whatsapp/telegram and other messaging apps.
a per topic mute for x hours/forever would be a nice feature for NTFY (at least for my use)
